File: ios/Sasquatsh/Services/AppearanceManager.swift
import SwiftUI

enum AppearanceMode: String, CaseIterable {
    case system, light, dark

    var colorScheme: ColorScheme? {
        switch self {
        case .system: return nil
        case .light: return .light
        case .dark: return .dark
        }
    }
}

@Observable
final class AppearanceManager {
    static let shared = AppearanceManager()

    var mode: AppearanceMode {
        didSet {
            UserDefaults.standard.set(mode.rawValue, forKey: "appearanceMode")
        }
    }

    var colorScheme: ColorScheme? {
        mode.colorScheme
    }

    private init() {
        let saved = UserDefaults.standard.string(forKey: "appearanceMode") ?? "system"
        mode = AppearanceMode(rawValue: saved) ?? .system
    }
}

File: android/app/src/main/java/com/sasquatsh/app/ui/theme/AppearanceManager.kt
package com.sasquatsh.app.ui.theme

import android.content.Context
import androidx.compose.runtime.compositionLocalOf
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

enum class AppearanceMode(val key: String) {
    SYSTEM("system"),
    LIGHT("light"),
    DARK("dark");

    companion object {
        fun fromKey(key: String): AppearanceMode =
            entries.find { it.key == key } ?: SYSTEM
    }
}

object AppearanceManager {
    private const val PREFS_NAME = "appearance"
    private const val KEY_MODE = "appearanceMode"

    private val _mode = MutableStateFlow(AppearanceMode.SYSTEM)
    val mode: StateFlow<AppearanceMode> = _mode

    fun init(context: Context) {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        val saved = prefs.getString(KEY_MODE, "system") ?: "system"
        _mode.value = AppearanceMode.fromKey(saved)
    }

    fun setMode(context: Context, mode: AppearanceMode) {
        _mode.value = mode
        context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            .edit()
            .putString(KEY_MODE, mode.key)
            .apply()
    }
}

val LocalAppearanceMode = compositionLocalOf { AppearanceMode.SYSTEM }
